syft/generic/abstract/hookable.py

```diff
@@ -0,0 +1,32 @@
+from functools import wraps
+
+
+def chain_call(obj, method_name, *args, **kwargs):
+    results = []
+    current = obj
+    while current is not None:
+        method = getattr(current, method_name, None)
+        if method:
+            results.append(method(*args, **kwargs))
+        current = current.child
+    return results
+
+
+def hookable(hookable_method):
```

Comment:
This does overlap with ObjectConstructor a bit - but I'm ok with it given that ObjectConstructor doesn't exist yet - and there's a good chance they'll actually be compatible pieces of functionality with each other (aka ObjectConstructor creates the method names which this decorator detects)